WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooSavedPackagesSelectionView.swift

```diff
@@ -97,21 +97,39 @@ struct WooSavedPackagesSelectionView: View {
 
     var body: some View {
         VStack(spacing: 0) {
-            Divider()
-            if viewModel.isLoadingPackages {
-                ProgressView()
-                    .progressViewStyle(.circular)
+            if !viewModel.hasSavedPackages {
+                // Show extra loading indicator in case there are no packages
+                if viewModel.isLoadingPackages {
+                    ProgressView()
+                        .progressViewStyle(.circular)
+                        .padding()
+                }
+                else {
+                    Button {
+                        viewModel.loadPackages()
+                    } label: {
+                        Image(systemName: "arrow.trianglehead.counterclockwise")
+                    }
                     .padding()
+                }
```

Comment:
Showing loading indicator only if we did not have any data before, and showing the refresh icon if we got no data from backend (so we can load packages manually since we do not have the list and pull to refresh in that case)

Reply:
Yep, we will need empty state views for various parts where we fetch backend data and we might not get any data (for various reasons)